### PR TITLE
Issue #10, and documentation

### PR DESCRIPTION
**Pull request by simon-brooke**
Hi

There are many things here which you may dislike; if you do I can submit a new pull request without them.

1. Generated documentation is included in the repository. I find this useful because you can use GitHub docs to auto-publish documentation (as, for example, I've done [here](https://simon-brooke.github.io/the-great-game/codox/index.html)). However there are very good arguments for saying that nothing that is generated should be in the repository.
2. I have linked to the JMonkeyEngine 3.4 javadocs. I've done this because it's the current stable, although I note that jme-clj is currently built against 3.3.2. Changing the documentation links to the 3.3.2 javadocs would be a single search and replace.
3. I've made the following changes to actual code:
    1. in `app-settings`, I've added `:dialog-image` as a key, and I've reordered the list of keys into alphabetical order;
    2. in `quat`, I've rewritten so that either three numbers, or to Quaternions and a number, can be passed as arguments, and I've written unit tests for this;
    3. I've rewritten `particle-emitter` to optionally accept keywords, but I have not written unit tests for this;
    4. I've rewritten `bullet-app-state` to handle more sophisticated broadphase algorithms, but I have not written unit tests for this.

Also, `defsimpleapp` has got reformatted. I apologise for this, it's just white-space change caused by the VSCode formatter.

### CODE REVIEW

**Comment by ertugrulcetin** [comment]
Wow, looks so nice! I'm curious what is your motivation to contribute to this project?
Also, are you going to keep contributing to the project?

File: project.clj [hunk 1]
```diff
@@ -2,6 +2,13 @@
 
 (defproject jme-clj "0.1.13"
 
+  :codox {:metadata    {:doc        "**TODO**: write docs"
+                        :doc/format :markdown}
+          :output-path "docs"
+          ;;:source-uri  "https://github.com/simon-brooke/jme-clj/blob/main/{filepath}#L{line}"
+          :source-uri   "file:///home/simon/workspace/jme-clj/{filepath}#L{line}"
```
Comment (ertugrulcetin) [inline project.clj:9]: We'd like to change the personal path :)

Comment (simon-brooke) [reply project.clj:9]: Oooops! Sorry, yes, of course!

**Comment by ertugrulcetin** [comment]
@simon-brooke what do you think about this ⬇️
>Wow, looks so nice! I'm curious what is your motivation to contribute to this project?
>Also, are you going to keep contributing to the project?

**Comment by simon-brooke** [comment]
My motivation to contribute is that I needed to use it, and in order to use it I needed documentation. I'd have needed to do this much work to be able to use it myself, and I think it's worth sharing so that other people don't have to do the same amount ot work.

I'm unlikely to keep contributing on this scale, but there are more things I'm planning to do.

**Comment by ertugrulcetin** [comment]
@simon-brooke thanks, good to know!

**Comment by ertugrulcetin** [comment]
Is PR ready? Did you fix the URI path?

**Comment by simon-brooke** [comment]
Sorry! Now done.

Note that line number offsets won't work until the master version on Github and the version from which the documentation was generated are the same.

**Pull request merged**